crates/burn-wgpu/src/compiler/wgsl/instructions.rs

```diff
@@ -248,7 +248,7 @@ impl Display for Instruction {
                 f.write_fmt(format_args!("{out} = {item}({lhs}[{rhs}]);\n"))
             }
             Instruction::Modulo { lhs, rhs, out } => {
-                f.write_fmt(format_args!("{out} = {lhs} % {rhs};\n"))
+                f.write_fmt(format_args!("{out} = (({lhs} % {rhs}) + {rhs}) % {rhs};\n"))
```

Comment:
Is this complicated formula there in order to support negatives and floats?
We have to keep in mind that this Modulo instruction is used in other kernels, mostly with simple unsigned ints, and often in the hot loop so we want to keep it as simple as possible. 
I think there is a difference between modulo which is the basic operator % and remainder which has to take negative and floats into account, and the remainder more complicated formula should be at the kernel level in burn-jit, not in the instruction.

Reply:
@louisfd yes, the `((a % b) + b) % b` form is there to keep the sign, I may have been a little over-zealous in matching the behavior of PyTorch. When you say that formula should be at the kernel level in burn-jit, do you mean it should implemented in `numeric.rs`, or do you mean something like it's own dedicated Operator::Remainder so there isn't that redirect to the pre-existing modulo operation?

It terms of what % means, in [Rust](https://stackoverflow.com/questions/31210357/is-there-a-modulus-not-remainder-function-operation) (and in WGSL), the % calculates the remainder, not the modulo. [Things get a bit wonky when negative numbers are used](https://stackoverflow.com/questions/13683563/whats-the-difference-between-mod-and-remainder). It's different in Python though hence the difference, and the need for PyTorch's [remainder](https://pytorch.org/docs/stable/generated/torch.remainder.html) operation.

Reply:
@louisfd I went ahead and created a custom instruction for the remainder operation so we avoid any clash with the original one.